specification/servicefabric/resource-manager/Microsoft.ServiceFabric/2017-07-01-preview/servicefabric.json

```diff
@@ -0,0 +1,3268 @@
+        },
+        "Weight": {
+          "$ref": "#/definitions/ServiceLoadMetricWeight"
+        },
+        "PrimaryDefaultLoad": {
+          "type": "integer",
+          "description": "Used only for Stateful services. The default amount of load, as a number, that this service creates for this metric when it is a Primary replica."
+        },
+        "SecondaryDefaultLoad": {
+          "type": "integer",
+          "description": "Used only for Stateful services. The default amount of load, as a number, that this service creates for this metric when it is a Secondary replica."
+        },
+        "DefaultLoad": {
+          "type": "integer",
+          "description": "Used only for Stateless services. The default amount of load, as a number, that this service creates for this metric."
+        }
+      },
+      "x-ms-sf-documentation": {
+        "Weight": {
+          "description": "The service load metric relative weight, compared to other metrics configured for this service, as a number."
+        }
+      }
+    },
+    "ServiceLoadMetricWeight": {
+      "type": "string",
+      "description": "Determines the metric weight relative to the other metrics that are configured for this service. During runtime, if two metrics end up in conflict, the Cluster Resource Manager prefers the metric with the higher weight. Possible values are following.\n\n  - Zero - Disables resource balancing for this metric. This value is zero.\n  - Low - Specifies the metric weight of the service load as Low. The value is 1.\n  - Medium - Specifies the metric weight of the service load as Medium. The value is 2.\n  - High - Specifies the metric weight of the service load as High. The value is 3.\n",
+      "enum": [
+        "Zero",
+        "Low",
+        "Medium",
+        "High"
+      ]
+    },
+    "ServicePlacementPoliciesList": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ServicePlacementPolicyDescription"
+      },
+      "description": "A list that describes the correlation of the service with other services."
+    },
+    "ServicePlacementPolicyDescription": {
+      "type": "object",
+      "discriminator": "Type",
+      "description": "Describes the policy to be used for placement of a Service Fabric service.",
+      "required": [
+        "Type"
+      ],
+      "properties": {
+        "Type": {
+          "$ref": "#/definitions/ServicePlacementPolicyType"
+        }
+      }
+    },
+    "ServicePlacementPolicyType": {
+      "type": "string",
+      "description": "The type of placement policy for a service fabric service. Following are the possible values.\n\n  - Invalid - Indicates the type of the placement policy is invalid. All Service Fabric enumerations have the invalid type. The value is zero.\n  - InvalidDomain - Indicates that th
```

Comment:
In application resource, you are refering the applicationTyperesource. It should be done using the arm resource id of the applicationTypeResource.